app/controllers/posts_controller.rb

```diff
@@ -1,10 +1,13 @@
 class PostsController < ApplicationController
+  include UniqueResourceUrl
+
   def index
     @posts = Post.includes(:comments, :categories).paginate(page: params[:page], per_page: 40)
   end
 
   def show
-    @post = Post.friendly.find(params[:id])
+    @post = post
+    ensure_or_redirect_to(post); return if performed?
```

Comment:
whoa... ugly line :/

Reply:
From the article you recommended earlier in the conversations(https://blog.arkency.com/2014/07/4-ways-to-early-return-from-a-rails-controller/), this struck me as the one that makes the most sense by just reading it.  
Out of curiosity, which one would recommend?

Reply:
```ruby
ensure_or_redirect_to(post) and return if performed?
```
does not work.

I could have sworn that I tried
```ruby
ensure_or_redirect_to(post)
return if performed?
```
and it errored, since I did not like the semicolon too much either. But it seems to work fine 🤷‍♂️